platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/ImportMigrationEntry.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Optional;
+
+/**
+ * The <code>ImportMigrationEntry</code> interfaces provides support for artifacts that are being
+ * imported during migration.
+ * <p>
+ * <b>
+ * This code is experimental. While this interface is functional
+ * and tested, it may change or be removed in a future version of the
+ * library.
+ * </b>
+ * </p>
+ */
+public interface ImportMigrationEntry extends MigrationEntry {
+    /**
+     * Retrieves a migration entry referenced from a property value defined in the properties file
+     * associated with this migration entry.
+     *
+     * @param name the name of the property that contains the reference to a migration entry
```

Comment:
❓  Some class level clarification of the migration entry mapping would be helpful. It sounds like, from the javadoc, that you specify an _individual_ property name that lives in some file, and by providing that, you get back the entry pertaining to that _file_ and not the property itself. This would imply there are multiple keys pointing to the same entry. 

Example:
```
ImportMigrationEntry systemPropertiesEntry = 
      getPropertyReferencedEntry("org.codice.ddf.system.hostname");

ImportMigrationEntry anotherReferenceToSystemPropertiesEntry = 
      getPropertyReferencedEntry("org.codice.ddf.system.protocol");
```
We would get the same entries because both keys live in the system properties file. 

Is this an accurate conceptualization? Or am I misunderstanding?